cloud-services/mbl-cloud-client/scripts/arm_update_activate.sh

```diff
@@ -194,6 +275,45 @@ if echo "${FIRMWARE_FILES}" | grep .ipk$; then
 
     exit "$?"
 
+elif BOOTLOADER_FILES=$(echo "${FIRMWARE_FILES}" | grep "${WKS_BOOTLOADER_FILENAME_RE}"); then
+
+    # ------------------------------------------------------------------------------
+    # Write one or more bootloader components from an update payload file to disk
+    # ------------------------------------------------------------------------------
+
+    # variable is unquoted to remove carriage returns, tabs, multiple spaces etc
+    for bl_file in $BOOTLOADER_FILES; do
```

Comment:
There should just be a single file name on each line, right? I suggest using something like
```Shell
printf "%s\n" "$BOOTLOADER_FILES" | while read bl_file; do
```
Functionally, I don't think it would make much difference unless you had files with whitespace in the names but it also more clearly shows that you're reading a filename from each line.

Reply:
Ah, be careful with piping into `while` and then using `exit` in the body of the loop though - the pipe will cause it to run in a subshell so the `exit`s will only happen in the subshell.

Maybe the `for` loop you had is better... Does `$IFS` apply to `for` loops in the busybox shell we're using? It does in Bash, I think, but I don't see anything about it for POSIX Shell at https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_05_03... I was thinking that setting `IFS` to `\n` for the `for` loop might be good but maybe it won't do anything.

You could also avoid the subshell for the `while` loop by doing something like
```Shell
while read bl_file; do
    # ...
done <<EOF
$BOOTLOADER_FILES
EOF
```
but that's harder to understand because you don't see the input to `while read` until the end of the loop body so I don't think I recommend it in this case.

Reply:
Resolving because I don't think there's anything to do.

Reply:
I don't think IFS applies to `for` loops in the busybox shell (which I think is `ash`) we're using. I think my for loop is ok here.